src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphEntityDAO.scala

```diff
@@ -39,6 +39,11 @@ class GraphEntityDAO extends EntityDAO with GraphDAO {
   }
 
   private def createVertex(workspace: Vertex, workspaceNamespace: String, workspaceName: String, entity: Entity, txn: RawlsTransaction): Vertex = txn withGraph { db =>
+    // check for illegal dot characters
+    validateUserDefinedString(entity.entityType)
+    validateUserDefinedString(entity.name)
+    entity.attributes.keys.foreach(validateUserDefinedString)
+
```

Comment:
I went and tested it - dotted names are fine but dotted types are not. In fact, edge labels in general cannot have dots. Try going into ```GraphDAO``` and adding a dot to one of the edge types. Lots of unit tests will break. Maybe this has to do with the fact that edges are stored as properties on vertices. It still fails if ```useLightweightEdges``` is disabled, though.

So it sounds like we should validate all edge labels, then?